### PR TITLE
REF: share IntervalIndex._check_method with CategoricalIndex

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/category.py [hunk 1]
```diff
@@ -704,3 +696,18 @@ def _delegate_method(self, name: str, *args, **kwargs):
         if is_scalar(res):
             return res
         return CategoricalIndex(res, name=self.name)
+
+
+def check_indexing_method(self, method):
```
Comment (jreback) [inline pandas/core/indexes/category.py:701]: why aren't you using pandas.core.missing.clean_fill_method?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/category.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/category.py:701]: because thats checking that the method is valid, whereas this is raising whenever the method is non-None.

Comment (jreback) [reply pandas/core/indexes/category.py:701]: ok seems an odd place to put this inside a sub-class of index, i would locate elsewhere, maybe pandas.core.indexes.api (which btw we should rename to something better)

Comment (jbrockmendel) [reply pandas/core/indexes/category.py:701]: its definitely odd.  we could put it in ExtensionIndex and then have DatetimeLikeIndex override it

Comment (jreback) [reply pandas/core/indexes/category.py:701]: ok thats a bit better place

Comment (jbrockmendel) [reply pandas/core/indexes/category.py]: updated+green

We're about to simplify the tar out of CategoricalIndex indexing

**Review by jreback — APPROVED** [review]

**Pull request merged**